Add a Custom CA Certificate to the Trusted Root Certificate Store (#1415)
astronomer/docs: astro/develop-project.md

</TabItem>
</Tabs>

## Add a CA certificate to an Astro Runtime image

> To allow Astro to authenticate with external services, it is sometimes necessary to add a custom, internal certificate authority (CA) certificate to the Astro Runtime image.

Reviewer: Can you check with tech reviewers about whether this would apply on Software too? We might need to add it to the "Customize image" doc there.

Reviewer 2: Correct, this applies anywhere the image is used (its specific to the image, though technically the same instructions would work for ap-airflow images, since that is being deprecated we don't have to document it there)

Author: @jwitz I added the procedure to the software documentation (Customize your image on Astronomer Software).

> To allow Astro to authenticate with external services, it is sometimes necessary to add a custom, internal certificate authority (CA) certificate to the Astro Runtime image.

Reviewer 2: ```suggestion
If you need your Astro deployment to communicate a securely with a remote service using a certificate signed by an untrusted/internal certificate authority (CA) you will need to add the CA certificate to the trust store inside the image.
```

1. In your Astro project `Dockerfile`, add the following entry below the existing `FROM` statement which specifies your Astro Runtime image version:

    ```docker

>     COPY gaw.crt /usr/local/share/ca-certificates/gaw/
>     RUN update-ca-certificates
>     USER astro

Reviewer 2: `gaw` here is short for `goanywhere`, and was in relation to the customer we were trailing this on.

```suggestion
    USER root
    COPY internal-ca.crt /usr/local/share/ca-certificates/my-company/
    RUN update-ca-certificates
    USER astro
```

    ```

> 2. Optional. Add additional `COPY` statements for each CA certificate your organization is using for external access.

Reviewer 2: ```suggestion
2. Optional. Add additional `COPY` statements before the `RUN update-ca-certificates` stanza for each CA certificate your organization is using for external access.
```

> 3. Save your changes and run the following command to rebuild your project:

Reviewer: This is only for running locally, while the setup is for allowing Astro to authenticate. Should we just remove this step and let people decide whether they need to test locally or deploy to a test Deployment?

Author: @jwitz I modified the third step of the procedure to address your concerns.
